contracts/FIFOMarket.sol

```diff
@@ -242,6 +283,51 @@ contract FIFOMarket is
     _removal.burnBatch(address(this), batchedIds, batchedAmounts);
   }
 
+  // TODO batch version of this?
+  function reserveRemoval(uint256 removalId) external returns (bool) {
```

Comment:
One thing I keep thinking about is whether or not a function like this should actually just transfer the removal back to the supplier. Have we thought through this or discussed this already?